### PR TITLE
Allow locating the keyname file in multiple valid locations.

### PR DESCRIPTION
**Pull request by tmarballi**
https://ocd.appscale.com:8080/job/Daily%20Build/1320/

### CODE REVIEW

File: lib/local_state.py [hunk 1]
```diff
@@ -280,8 +284,12 @@ def get_key_path_from_name(cls, keyname):
     Returns:
       A str that indicates where the private key can be found.
     """
-    return cls.LOCAL_APPSCALE_PATH + keyname + ".key"
-
+    for local_path in cls.VALID_KEY_PATHS:
+      key_file_path = local_path + keyname + ".key"
+      if (os.path.isfile(key_file_path)):
```
Comment (menivaitsi) [inline lib/local_state.py]: Change this line to: 
`if os.path.isfile(key_file_path):`

The external parentheses are not needed.

File: lib/local_state.py [hunk 2]
```diff
@@ -280,8 +284,12 @@ def get_key_path_from_name(cls, keyname):
...
+        return key_file_path
+      else:
```
Comment (menivaitsi) [inline lib/local_state.py]: Remove the else part. It's a no op.

File: test/test_appscale_run_instances.py [hunk 3]
```diff
@@ -395,8 +395,12 @@ def test_appscale_in_one_node_virt_deployment(self):
 
   def test_appscale_in_one_node_cloud_deployment_auto_spot_price(self):
     # let's say that appscale isn't already running
+    LOCAL_APPSCALE_PATH = os.path.expanduser("~") + os.sep + ".appscale" + \
```
Comment (menivaitsi) [inline test/test_appscale_run_instances.py]: Avoid all caps for local variables.

File: lib/local_state.py [hunk 4]
```diff
@@ -45,6 +45,10 @@ class LocalState(object):
   # AppScale deployment metadata.
   LOCAL_APPSCALE_PATH = os.path.expanduser("~") + os.sep + ".appscale" + os.sep
 
+  ETC_APPSCALE_KEY_PATH = "/etc/appscale/keys/cloud1/"
```
Comment (menivaitsi) [inline lib/local_state.py]: Add comments to describe these constants.

**Comment by tmarballi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1337/

**Comment by tmarballi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1341/

**Pull request merged**